static_frame/core/quilt.py

```diff
@@ -426,15 +426,25 @@ def _update_axis_labels(self) -> None:
                     init_exception_cls=ErrorInitQuilt,
                     )
 
+        axis_label = 'index' if self._axis == 0 else 'column'
+        axis_labels = 'indices' if self._axis == 0 else 'columns'
+        err_msg = f"Duplicate {axis_label} labels across frames. Either ensure all {axis_labels} are unique for all frames, or set retain_labels=True to obtain an IndexHierarchy"
```

Comment:
I like the improved `Exception` message this provides, but maybe this should be tucked away in a function only called when the `Exception` is raised. Always doing it for non-exceptional cases seems unnecessary.